### ISSUE TITLE
Neural Architecture Search with Reinforcement Learning

### ISSUE THREAD

**Issue by leo-p** [leo-p/papers#26]
https://arxiv.org/pdf/1611.01578.pdf

Neural networks are powerful and flexible models that work well for many difficult learning tasks in image, speech and natural language understanding. Despite their success, neural networks are still hard to design. In this paper, we use a recurrent network to generate the model descriptions of neural networks and train this RNN with reinforcement learning to maximize the expected accuracy of the generated architectures on a validation set. On the CIFAR-10 dataset, our method, starting from scratch, can design a novel network architecture that rivals the best human-invented architecture in terms of test set accuracy. Our CIFAR-10 model achieves a test error rate of 3.65, which is 0.09 percent better and 1.05x faster than the previous state-of-the-art model that used a similar architectural scheme. On the Penn Treebank dataset, our model can compose a novel recurrent cell that outperforms the widely-used LSTM cell, and other state-of-the-art baselines. Our cell achieves a test set perplexity of 62.4 on the Penn Treebank, which is 3.6 perplexity better than the previous state-of-the-art model. The cell can also be transferred to the character language modeling task on PTB and achieves a state-of-the-art perplexity of 1.214.

**Comment by leo-p** [comment]
## Summary:
- *Dataset:* [CIFAR10](https://www.cs.toronto.edu/~kriz/cifar.html) and [PTB](https://catalog.ldc.upenn.edu/ldc99t42).
- *Objective:* Design a network that will itself find the best architecture for a given task.

## Inner-workings:
The meta-network (a RNN) generates a string specifying the child network parameters. Such a child network is then trained for 35-50 epochs and its accuracy is used as the reward to train the meta-network with Reinforcement Learning.
The RNN first generates networks with few layers (6) then this number is increased as training progresses.

## Architecture:
They develop one architecture for CNN where they predict each layers characteristic plus it's possible skip-connection:
<img width="600" alt="screen shot 2017-05-24 at 8 13 01 am" src="https://cloud.githubusercontent.com/assets/17261080/26389176/d807de42-4058-11e7-942a-8a129558e126.png">

And one specific for LTSM-style:
<img width="600" alt="screen shot 2017-05-24 at 8 13 26 am" src="https://cloud.githubusercontent.com/assets/17261080/26389190/e2bfd506-4058-11e7-9168-62abd040156e.png">

## Distributed setting:
Bellow is the distributed setting that they use with parameter servers connected to replicas (GPUs) that trained child networks.
<img width="400" alt="screen shot 2017-05-24 at 8 09 05 am" src="https://cloud.githubusercontent.com/assets/17261080/26389084/5e354456-4058-11e7-83a9-089cb2c115b7.png">

## Results:
Overall they trained 12800 networks on 800 GPUs but they achieve state of the art results which not human intervention except the vocabulary selection (activation type, type of cells, etc). Next step, transfer learning from one task to another for the meta-network?
